agents-common/src/main/java/org/apache/ranger/plugin/policyengine/RangerPolicyEngineImpl.java

```diff
@@ -83,6 +84,53 @@ public RangerPolicyEngineImpl(ServicePolicies servicePolicies, RangerPluginConte
         policyEngine     = new PolicyEngine(servicePolicies, pluginContext, roles, isUseReadWriteLock);
         serviceConfig    = new ServiceConfig(servicePolicies.getServiceConfig());
         requestProcessor = new RangerDefaultRequestProcessor(policyEngine);
+
+        Map<String, String> serviceConfigMap = servicePolicies.getServiceConfig();
```

Comment:
can the whole pluginContext setting be done upfront in basePlugin before RangerPolicyEngineImpl is created?